toffy/tiling_utils.py

```diff
@@ -1271,11 +1273,311 @@ def generate_validation_annot(manual_to_auto_map, manual_auto_dist, check_dist=2
+    fov_width = fov_top_right_pixels[1] - fov_top_left_pixels[1]
+
+    # draw the rectangle defining this fov
+    # NOTE: the x-y axis for rectangles introduces yet another coordinate axis system,
+    # so we'll need to reverse the order fov_top_left_pixels is passed in
+    fov_top_left_pixels_rect = tuple(reversed(fov_top_left_pixels))
+    dr = FOVRectangle(
+        fov_top_left_pixels_rect, fov_width, fov_height, fov_color, fov_info['name'], ax
+    )
+
+    return dr
+
+
+def delete_tiled_region_fovs(rectangles, tiled_region_fovs):
+    """Delete all the FOVs from tiled_region_fovs with lindwidth 5 (indicating its been selected)
+
+    Helper function to `delete_fovs` in `tiled_region_interactive_remap`
+
+
+    Args:
+        rectangles (dict):
+            Maps each FOV to its corresponding rectangle instance
+        tiled_region_fovs (dict):
+            The list of FOVs to overlay for each tiled region
+    """
+
+    # define a list of FOVs to delete
+    # NOTE: only FOVs with a linewidth of 5, indicates user has selected it
+    fovs_delete = [fov for fov in rectangles if rectangles[fov].rect.get_linewidth() == 5]
+
+    # overwrite the list of FOVs in tiled_region_fovs to only contain FOVs not in fov_delete
+    tiled_region_fovs['fovs'] = [
+        fov for fov in tiled_region_fovs['fovs'] if fov['name'] not in fovs_delete
+    ]
+
+    # now delete the corresponding rectangle for each FOV in fovs_delete
+    for fov in fovs_delete:
+        rectangles[fov].rect.remove()
+        del rectangles[fov]
+
+
+# TODO: potential type hinting candidate?
+def tiled_region_interactive_remap(tiled_region_fovs, tiling_params, slide_img, tiled_region_path,
+                                   figsize=(7, 7)):
+    """Creates the tiled region interactive interface for manual to auto FOVs
+
+    Args:
+        tiled_region_fovs (dict):
+            The list of FOVs to overlay for each tiled region
+        tiling_params (dict):
+            The tiling parameters generated for each tiled region
+        slide_img (numpy.ndarray):
+            The image to overlay
+        tiled_region_path (str):
+            The path to the file to save the tiled regions to
+        figsize (tuple):
+            The size of the interactive figure to display
+    """
+
+    # error check: ensure mapping path exists
+    if not os.path.exists(os.path.split(tiled_region_path)[0]):
+        raise FileNotFoundError(
+            "Path %s to tiled_region_path does not exist, "
+            "please rename to a valid location" % os.path.split(tiled_region_path)[0]
+        )
+
+    # if there isn't a coreg_path defined, the user needs to run update_coregistration_params first
+    if not os.path.exists(os.path.join('..', 'toffy', 'coreg_params.json')):
+        raise FileNotFoundError(
+            "You haven't co-registered your slide yet. Please run "
+            "update_coregistraion_params.ipynb first."
```

Comment:
This is now the 1_set_up_toffy notebook. Can you also adjust the error message in the TMA script?